packages/foundry/script/Deploy.s.sol

```diff
@@ -3,35 +3,33 @@ pragma solidity ^0.8.19;
 
 import "../contracts/YourContract.sol";
 import "./DeployHelpers.s.sol";
+import { DeployYourContract } from "./00_deploy_your_contract.s.sol";
 
 contract DeployScript is ScaffoldETHDeploy {
+  uint256 deployerPrivateKey;
+
   error InvalidPrivateKey(string);
 
-  function run() external {
-    uint256 deployerPrivateKey = setupLocalhostEnv();
+  constructor() {
+    deployerPrivateKey = setupLocalhostEnv();
+  }
+
+  function run() external checkDeployerPrivateKey deploymentExporter {
```

Comment:
i think we can have one single modifier `ScaffoldEthDeployerRunner` that does :
```solidity
checkDeployerPrivateKey();
_;
exportDeployments();
```

instead of having two, this is less error prone, in case users forget to add one of them. @technophile-04